common/service/service.go

```diff
@@ -116,6 +119,10 @@ func (h *serviceImpl) Start() {
 	h.metricsScope.Counter(metrics.RestartCount).Inc(1)
 	h.runtimeMetricsReporter.Start()
 
+	if err := h.pprofInitializer.Initialize(); err != nil {
```

Comment:
You probably want to allow service startup if the pprof initializer is not set.

Reply:
fixed